Sources/Classes/core/Downloader/MiniAppClient.swift

```diff
@@ -148,39 +148,6 @@ internal class MiniAppClient: NSObject, URLSessionDownloadDelegate {
         return requestDataFromServer(urlRequest: urlRequest, completionHandler: completionHandler)
     }
 
-    func requestFromServer(urlRequest: URLRequest, retry500: Int = 0, completionHandler: @escaping (Result<ResponseData, Error>) -> Void) {
-        return session.startDataTask(with: urlRequest) { (result) in
-            switch result {
-            case .success(let responseData):
-                let statusCode = responseData.httpResponse.statusCode
-                let logIcon = statusCode < 300 ? "🟢" : "🟠"
-                MiniAppLogger.d("[\(statusCode)] urlRequest \(urlRequest.url?.absoluteString ?? "-") : \n\(String(data: responseData.data, encoding: .utf8) ?? "Empty response")", logIcon)
-                responseData.httpResponse.allHeaderFields.forEach { key, value in  MiniAppLogger.d("[\(key)]\t \(value)", "\t🎩")}
-
-                if !(200...299).contains(statusCode) {
-                    let failure = self.handleHttpResponse(responseData: responseData.data, httpResponse: responseData.httpResponse)
-                    if statusCode >= 500, retry500 < 5 {
-                        let backOff = 2.0
-                        let retry = retry500 + 1
-                        let waitTime = 0.5*pow(backOff, Double(retry500))
-                        let failureMessage = "\(failure.localizedDescription) : Attempt [\(retry)]."
-                        MiniAppLogger.d("\(failureMessage) \nRetry in \(waitTime)s", "🟠")
-                        return DispatchQueue.main.asyncAfter(deadline: .now() + waitTime) {
-                            self.requestFromServer(urlRequest: urlRequest, retry500: retry, completionHandler: completionHandler)
-                        }
-                    }
-                    MiniAppLogger.d("\(failure.localizedDescription)", "🔴")
-                    return completionHandler(.failure(failure))
-                }
-                return completionHandler(.success(ResponseData(responseData.data,
-                                                               responseData.httpResponse)))
-            case .failure(let error):
-                MiniAppLogger.d("urlRequest \(urlRequest.url?.absoluteString ?? "-") : Failure", "🔴")
-                return completionHandler(.failure(error))
-            }
-        }
-    }
```

Comment:
Never noticed we had this in double!

Reply:
@Climbatize Not sure about other refactorings, but I have used the following,
`typealias MAResponseDataHandler = (Result<ResponseData, MASDKError>) -> Void`
This is used in many places.